tests/main/views/test_users.py

```diff
@@ -501,22 +503,23 @@ def test_return_400_for_invalid_user_json(self):
 
         assert response.status_code == 400
         data = json.loads(response.get_data())["error"]
-        assert "JSON was not a valid format" in data
+        assert data == "JSON was not a valid format. u'' is too short"
 
-    def test_return_400_for_invalid_user_role(self):
+    @mock.patch('app.db.session.commit')
+    def test_create_user_catches_db_errors(self, db_commit):
+        db_commit.side_effect = DataError("Unable to commit", orig=None, params={})
         response = self.client.post(
             '/users',
             data=json.dumps({
                 'users': {
-                    'emailAddress': 'joeblogs@email.com',
-                    'password': '0000000000',
-                    'role': 'invalid',
+                    'emailAddress': 'joeblogs@email.gov.uk',
+                    'phoneNumber': '01234 567890',
+                    'password': '1234567890',
+                    'role': 'buyer',
                     'name': 'joe bloggs'}}),
             content_type='application/json')
-
         assert response.status_code == 400
-        data = json.loads(response.get_data())["error"]
-        assert "JSON was not a valid format" in data
+        assert json.loads(response.get_data())["error"] == "Invalid user role"
```

Comment:
I was wondering about that. That particular `try...except` aborts with that error message in the code. I was a bit loathe to change actual code in this PR rather than just tests, but if the message should be `Database Error:...` I can change it?